app/controllers/propertyLinking/ClaimPropertyOccupancyController.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.propertyLinking
+
+import actions.AuthenticatedAction
+import actions.propertylinking.WithLinkingSession
+import com.google.inject.Singleton
+import config.ApplicationConfig
+import connectors.SubmissionIdConnector
+import connectors.propertyLinking.PropertyLinkConnector
+import controllers._
+import form.Mappings._
+import models._
+import play.api.Configuration
+import play.api.data.Form
+import play.api.data.Forms._
+import play.api.i18n.{Messages, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepo
+import services.BusinessRatesAttachmentsService
+import services.propertylinking.PropertyLinkingService
+import uk.gov.hmrc.http.UpstreamErrorResponse
+import uk.gov.hmrc.propertylinking.errorhandler.CustomErrorHandler
+import uk.gov.voa.play.form.ConditionalMappings._
+import views.helpers.Errors
+import java.time.LocalDate
+
+import javax.inject.{Inject, Named}
+import org.checkerframework.checker.compilermsgs.qual.CompilerMessageKey
+import utils.Formatters
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ClaimPropertyOccupancyController @Inject()(
+      val errorHandler: CustomErrorHandler,
```

Comment:
this controller seems to have a ton of things injected into its constructor. Does it really need all this?
```
    val submissionIdConnector: SubmissionIdConnector,
    val propertyLinksConnector: PropertyLinkConnector,
    val runModeConfiguration: Configuration,
```
 can be removed